packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -114,17 +117,29 @@ class BottomAppBar extends StatefulWidget {
   /// If this property is null then [BottomAppBarTheme.shape] of
   /// [ThemeData.bottomAppBarTheme] is used. If that's null then the shape will
   /// be rectangular with no notch.
+  ///
+  /// Ignored if the ambient [ThemeData.useMaterial3] is true; the Material 3
+  /// spec defines the shape without a notch:
+  ///   https://m3.material.io/components/bottom-app-bar/specs#9d91e4a4-a6d9-4ad5-afad-430267ebb4e9
   final NotchedShape? shape;
 
   /// {@macro flutter.material.Material.clipBehavior}
   ///
   /// Defaults to [Clip.none], and must not be null.
+  ///
+  /// Ignored if the ambient [ThemeData.useMaterial3] is true; the Material 3
+  /// spec defines the shape without a notch:
+  ///   https://m3.material.io/components/bottom-app-bar/specs#9d91e4a4-a6d9-4ad5-afad-430267ebb4e9
   final Clip clipBehavior;
```

Comment:
Can this be simplified like so?

```suggestion
  ///
  /// Has no effect if there is no notch; see [shape].
  final Clip clipBehavior;
```

Similarly for [notchMargin].

(And if so, then that seems like a good docs improvement even before this PR's changes to the behavior of the code.)